### PR TITLE
created day-night-theme reusable component

### PR DESCRIPTION
**Pull request by olgakahraman**
I created a toggle-button element and added css to display it like a toggle button, i added it to navbar, and added js for changing background color if button is toggled. I imported different css of components which background I want to be changed if btn toggled. i removed previous day-theme classNames and I added .dark classname for elements which are supposed to change backgroud

### CODE REVIEW

File: src/components/BookCard/BookCard.jsx
Comment (ElijahMSmith) [inline src/components/BookCard/BookCard.jsx]: When I run this code, it crashes saying that `publishDate` on this line is undefined.

File: src/components/NavBar/Navbar.jsx
Comment (ElijahMSmith) [inline src/components/NavBar/Navbar.jsx:30]: This is not formatted correctly. Apart from that, clicking a theme toggle should not send me to a different page. If I had written some part of a comment on a book discussion and all that got lost, I would be really upset!

File: src/components/BookCard/BookCard.css
Comment (ElijahMSmith) [inline src/components/BookCard/BookCard.css]: I don't think we should be overwriting this many stypes in the dark version of a component (especially something like display). Remember that `.card` also provides some styles to the component that aren't lost, just overwritten by those here. Things like layout, text format/size, etc. should probably be the same light or dark. Things like color, border, etc. should be the things overwritten by these styles.

File: src/components/BookDetails/BookDetails.css
Comment (ElijahMSmith) [inline src/components/BookDetails/BookDetails.css]: We don't need a newline here.

**Review by ElijahMSmith — CHANGES_REQUESTED** [review]
I don't think this should be a priority right now. That said, you've already written some code here so as long as we don't spend very long on this I am willing to finish out this MR.

If we do stick with this, I would suggest we use CSS variables as described [here](https://blog.logrocket.com/dark-mode-react-in-depth-guide/) and specify our CSS rules to use these variables.

We might start thinking about the set of colors we want to use across our site. There's a thing called the `60/30/10` rule (https://www.flowmapp.com/blog/qa/60-30-10-rule) that we could decide as a group which colors we want to use for backgrounds and for text to contrast on those backgrounds.

File: src/components/BookCard/BookCard.jsx
Comment (olgakahraman) [reply src/components/BookCard/BookCard.jsx]: yes this i already changed in another branch. there is supposed to be publishedDate apart of publishDate.

File: src/App.css
Comment (ElijahMSmith) [inline src/App.css:2]: Please run a formatter on this page!

File: package.json
Comment (ElijahMSmith) [inline package.json:1]: The `use-persisted-state` dependency does not work with React v18 or higher. If you delete your `node_modules` and `npm i` you will get an error and not be able to install the dependencies.

Additionally, we should try to not bump react versions. It will probably be OK, but due to changes between major version (18->19 here) in a really big library or language we might encounter some new bugs introduced or some unexpected changes that are problematic.

I think you should keep `react`, `react-dom`, and `react-router-dom` at their old versions and only install `react-responsive`. We've used `localStorage` before in the React class, we don't need a library for that.

File: src/components/day-night-theme/DayNightTheme.css
Comment (ElijahMSmith) [inline src/components/day-night-theme/DayNightTheme.css]: Nitpick: I think these should go in `App.css` so they are easier to find if someone is looking for the default colors.

**Review by ElijahMSmith — APPROVED** [review]
Please make sure your code works before submitting, but I am approving so that we don't have to keep going back and forth.

**Comment by ElijahMSmith** [comment]
(Also please squash your commits before you merge in)

**Pull request merged**
